elixir/apps/api/lib/api/gateway/channel.ex

```diff
@@ -110,38 +110,66 @@ defmodule API.Gateway.Channel do
       when gateway_id == id do
     tuple = {flow.client_id, flow.resource_id}
 
-    socket =
-      if flows_map = Map.get(socket.assigns.flows, tuple) do
-        flow_map = Map.delete(flows_map, flow.id)
+    if flows_map = Map.get(socket.assigns.flows, tuple) do
+      flow_map = Map.delete(flows_map, flow.id)
+      remaining = map_size(flow_map)
 
-        with 0 <- map_size(flow_map),
-             {:ok, new_flow} <- Flows.reauthorize_flow(flow) do
-          flow_map = %{
-            new_flow.id => new_flow.expires_at
-          }
+      if remaining == 0 do
+        case Flows.reauthorize_flow(flow) do
+          {:ok, new_flow} ->
+            flow_map = %{
+              new_flow.id => new_flow.expires_at
+            }
 
-          push(
-            socket,
-            "access_authorization_expiry_updated",
-            Views.Flow.render(new_flow, new_flow.expires_at)
-          )
+            Logger.info("Updated flow authorization",
+              account_id: flow.account_id,
+              client_id: flow.client_id,
+              resource_id: flow.resource_id,
+              new_flow_id: new_flow.id
```

Comment:
Would including the old flow ID in this message help with potential debugging ever?

Reply:
👍